vaadin-grid-flow-testbench/src/main/java/com/vaadin/flow/component/grid/testbench/GridElement.java

```diff
@@ -32,6 +32,18 @@
 @Element("vaadin-grid")
 public class GridElement extends TestBenchElement {
 
+    protected void waitUntilLoadingFinished() {
+        waitUntil(e -> !isLoading());
+    }
+
+    protected boolean isLoading() {
+        return (Boolean) executeScript("return arguments[0]._cache.isLoading()",
+                this)
+                || (Boolean) executeScript(
+                        "return arguments[0].$connector.hasRootRequestQueue();",
+                        this);
```

Comment:
I wonder if this can be done within a single `executeScript` call and if it's worth changing it.

Reply:
Good point. Fixed in both GridElement and TreeGridElement.